include/tscpp/util/cmp_op.h

```diff
@@ -0,0 +1,164 @@
+/** @file
+
+   Convenient definition of comparison operators for user-defined types.
+
+   @section license License
+
+   Licensed to the Apache Software Foundation (ASF) under one
+   or more contributor license agreements.  See the NOTICE file
+   distributed with this work for additional information
+   regarding copyright ownership.  The ASF licenses this file
+   to you under the Apache License, Version 2.0 (the
+   "License"); you may not use this file except in compliance
+   with the License.  You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+ */
+
+/*
+Suppose that, for types T1 and T2, there exists a compare function with return type int.  The two parameters to the
+function have types const T1 & and const T2 &.  The function returns a negative value if the first parameter is less
+than the second parameter, 0 if the first parameter is equal to the second parameter, and a positive value if the first
+parameter is greater than the second parameter.  This include file allows the definition of the comparison operators
+'operator R (const T1 &, const T2 &)' for R in ==, !=, <, <=, >, >= .  If T1 and T2 are not the same type, the
+comparison operators 'operator R (const T2 &, const T1 &)' will also be defined.  To define the comparison operators,
+provide a specialization of the ts::cmp_op::Enable like this:
+
+namespace ts
+{
+namespace cmp_op
+{
+template <>
+struct Enabled<T1, T2> : public Yes<T1, T2, compare_function> {};
```

Comment:
I have to argue about naming again. I'd prefer "Operator" or "Compare" instead of "Enabled", and use "Enabled" instead of "Yes". I think that would be clearer in use. E.g.
```
template <>
struct Compare<T1,T2> : public Enabled<T1, T2, cmp> {};
```

Reply:
Either "Operator" or "Compare" seems redundant in a namespace called cmp_op.